### ISSUE TITLE
sql-open problems with postgres 10

### ISSUE THREAD

**Issue by glicerico** [opencog/atomspace#2134]
sql-open command fails if using postgres 10 with the following message:
```
Backtrace:
           6 (apply-smob/1 #<catch-closure 7f466c9d41e0>)
In ice-9/boot-9.scm:
    705:2  5 (call-with-prompt _ _ #<procedure default-prompt-handle…>)
In ice-9/eval.scm:
    619:8  4 (_ #(#(#<directory (guile-user) 7f46664c3140>)))
In ice-9/boot-9.scm:
   2312:4  3 (save-module-excursion _)
  3822:12  2 (_)
In launch-cogserver.scm:
     56:0  1 (_)
In unknown file:
           0 (opencog-extension sql-open ("postgres:///test"))

ERROR: In procedure opencog-extension:
Throw to key `C++-EXCEPTION' with args `("sql-open" "Failed to execute SQL command!\nPQresult message: ERROR:  column \"increment_by\" does not exist\nLINE 1: SELECT 1            - (SELECT increment_by FROM uuid_pool)\n                                      ^\nQUERY:  SELECT 1            - (SELECT increment_by FROM uuid_pool)\nCONTEXT:  PL/pgSQL function inline_code_block line 1 at assignment\n\nPQ query was: DO $$DECLARE    under BIGINT;BEGIN   IF (SELECT count(*) FROM pg_stat_activity WHERE           datname=(SELECT current_database())) = 14 THEN      under := 1            - (SELECT increment_by FROM uuid_pool);      IF (1 < under) THEN          RAISE NOTICE 'Set uuid_pool sequence to %', under;         PERFORM (SELECT setval('uuid_pool', under));      END IF;   END IF;END $$; (/home/andres/MyOpenCogSources/atomspace/opencog/persist/sql/multi-driver/ll-pg-cxx.cc:130)\nFunction args:\n(postgres:///test)")'.
```

From this site (and others) https://help.heroku.com/WKJ027JH/rails-error-after-upgrading-to-postgres-10 it seems like there was a change in the way id sequences work in postgres now.
If not an easy fix, we should specify that a previous postgres version should be used.

**Comment by linas** [comment]
There are two possible fixes: one is compile-time static, another would be dynamic.  The static fix would be
```
#ifdef POSTGRES_10
...
#else
...
#endif
```
somewhere near line 176 of `opencog/persist/sql/multi-driver/SQLUUID.cc`  That's more-or-less the easy fix.  The dynamic fix is to somehow detect at run-time whether one is on postgres 9 or 10, and then just do a run-time if-then-else. 

Either fix is easy, viz an afternoon putzing around with postgres, and reading docs, if you know postgres already. If you don't know SQL at all, then probably a minimum of a few days, or a week learning SQL+postgres quirks.  The problem here is that the lines 166-182 are defining a postgres stored procedure, which is fairly advanced, arcane, cryptic and totally non-standard SQL. Lots of  "read the docs very very carefully" type research, so if you're an SQL newbie, it would be pretty confusing. Its confusing even if you're not a beginner.

**Comment by linas** [comment]
I could maybe suggest a specific outlined fix, but I don't want to test it.

**Comment by linas** [comment]
Huh. I was looking at a very old branch when I wrote the above. Looks like the ifdef version is already implemented at lines 180-185  and for some reason, your compiled version did not pick that up. I am changing this to run-time query, now, please wait for pull req.

**Comment by linas** [comment]
So I think #2135 fixes this. @glicerico I think this fixes the bug, but I can't test. Please test; if it passes, please merge. If it fails, can you fix whatever is failing about it?

**Comment by glicerico** [comment]
Issue fixed with pull requests #2136 and #2137
